users/serializers.py

```diff
@@ -1,72 +1,148 @@
-import datetime
+import re 
+
+from django.contrib.auth.password_validation import validate_password
+from django.core import validators
+from django.utils.translation import gettext_lazy as _
 
 from rest_framework import serializers
 
-from .models import UserAccount
+from users.models import Account
 
 
-class UserSerializer(serializers.ModelSerializer):
-    active = serializers.BooleanField(source="is_active", default=True)
-    verified = serializers.BooleanField(source="is_verified", default=False)
-    superuser = serializers.BooleanField(source="is_superuser", default=False)
-    staff = serializers.BooleanField(source="is_staff", default=False)
 
-    class Meta:
-        model = UserAccount
-        fields = [
-            "id",
-            "email",
-            "username",
-            "password",
-            "active",
-            "verified",
-            "superuser",
-            "staff",
-            "created_time",
-            "last_login",
-        ]
-
-        extra_kwargs = {
-            "password": {"write_only": True},
-        }
 
+
+""" Serializers are initialised here  """
+
+class UserSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(required=True, write_only=True)
+    class Meta:
+        model = Account
+        exclude = ('id', 'is_active', 'is_superuser', 'is_staff', 'created_time', 'last_login', 'is_verified', 'groups', 'user_permissions')
+
+
+    def validate_email(self, email):
+        pattern = '[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+'
+        
+        if re.match(pattern, email):
+            if Account.objects.filter(email=email).exists():
+                raise serializers.ValidationError('Email already exists!')
+            return email
+        raise serializers.ValidationError('Invalid email format!')
+    
+    
     def create(self, validated_data):
-        validated_data["is_active"] = True
-        user = UserAccount.objects.create_user(**validated_data)
+        password = self.validated_data.get('password')
+        # pattern = r'!@#$%&\^*(),.?":{}|<>'
+        
+        error_messages = []
+        
+        # if not re.search(pattern, password):
+        #         error_messages.append(f'Passwords must contain at least one SPECIAL characters {pattern}')
+            
+        try: 
+            validate_password(password)   
```

Comment:
Validation of passwords should be done in the `validate_password` method and not in the create method. There you can raise a `ValidationError(message)` directly